kratos/linear_solvers/linear_solver.h

```diff
@@ -257,48 +254,115 @@ class LinearSolver
+    /**
+     * @brief This method checks if the dimensions of the system of equations are consistent (dense matrix for RHS and unknowns version)
+     * @param rA The LHS of the system of equations
+     * @param rX The matrix containing the unkowns
+     * @param rB The matrix containing the RHSs of the system of equations
+     * @return True if consistent, false otherwise
+     */
+    virtual bool IsConsistent(
+        SparseMatrixType& rA, 
+        DenseMatrixType& rX,
+        DenseMatrixType& rB
+        ) 
     {
         const SizeType size = TSparseSpaceType::Size1(rA);
-
-        return ((size ==  TSparseSpaceType::Size2(rA)) &&
-                (size ==  TDenseSpaceType::Size1(rX)) &&
-                (size ==  TDenseSpaceType::Size1(rB)) &&
-                (TDenseSpaceType::Size2(rX) == TDenseSpaceType::Size2(rB)));
+        const SizeType size_a = TSparseSpaceType::Size2(rA);
+        const SizeType size_1_x = TDenseSpaceType::Size1(rX);
+        const SizeType size_1_b = TDenseSpaceType::Size1(rB);
+        const SizeType size_2_x = TDenseSpaceType::Size2(rX);
+        const SizeType size_2_b = TDenseSpaceType::Size2(rB);
+
+        KRATOS_WARNING_IF("LinearSolver", size !=  size_a) << "WARNING: Your LHS matrix is not square. Sizes: " << size << " vs " << size_a << std::endl ;
+        KRATOS_WARNING_IF("LinearSolver", size !=  size_1_x) << "WARNING: Your unkowns matrix is not properly sized. Sizes: " << size << " vs " << size_1_x << std::endl ;
+        KRATOS_WARNING_IF("LinearSolver", size !=  size_1_b) << "WARNING: Your RHS matrix is not properly sized. Sizes: " << size << " vs " << size_1_b << std::endl ;
+        KRATOS_WARNING_IF("LinearSolver", (size_2_x != size_2_b)) << "WARNING: Your RHS matrix and unkowns matrix are not consistently sized. The number of systems to solve is not consistent. Sizes: " << size_2_x << " vs " << size_2_b << std::endl ;
+        
+        return ((size ==  size_a) &&
+                (size ==  size_1_x) &&
+                (size ==  size_1_b) &&
+                (size_2_x == size_2_b));
     }
 
-
-    virtual bool IsNotConsistent(SparseMatrixType& rA, VectorType& rX, VectorType& rB)
+    /**
+     * @brief This method checks if the dimensions of the system of equations are not consistent
+     * @param rA The LHS of the system of equations
+     * @param rX The vector containing the unkowns
+     * @param rB The RHS of the system of equations
+     * @return False if consistent, true otherwise
+     */
+    virtual bool IsNotConsistent(
+        SparseMatrixType& rA, 
+        VectorType& rX, 
+        VectorType& rB
+        )
     {
         return (!IsConsistent(rA, rX, rB));
     }
-
-    virtual bool IsNotConsistent(SparseMatrixType& rA, DenseMatrixType& rX, DenseMatrixType& rB)
+    
+    /**
+     * @brief This method checks if the dimensions of the system of equations are not consistent (dense matrix for RHS and unknowns version)
```

Comment:
IMO this is wrong/misleading, the argument is a sparse matrix

```suggestion
     * @brief This method checks if the dimensions of the system of equations are not consistent
```

maybe omitting it?